src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AbstractGenerateNuGetPackageCommand.cs

```diff
@@ -23,18 +40,66 @@ protected AbstractGenerateNuGetPackageCommand(UnconfiguredProject unconfiguredPr
 
         protected override Task<CommandStatusResult> GetCommandStatusAsync(IProjectTree node, bool focused, string commandText, CommandStatus progressiveStatus) =>
             ShouldHandle(node) ?
-                GetCommandStatusResult.Handled(GetCommandText(), CommandStatus.Enabled) :
+                GetCommandStatusResult.Handled(GetCommandText(), IsReadyToBuild() ? CommandStatus.Enabled : CommandStatus.Supported) :
                 GetCommandStatusResult.Unhandled;
 
+        private bool IsReadyToBuild()
+        {
+            return ErrorHandler.Succeeded(_buildManager.QueryBuildManagerBusy(out int busy)) && busy == 0;
```

Comment:
Can you please use constants instead of magic numbers? ie, `VSConstants.S_OK` instead of 0.

Using `ErrorHandler.Succeeded` here is going to swallow any exceptions. Is that the behavior we want?